app/src/pages/inside/launchesPage/launchesPage.jsx

```diff
@@ -214,7 +214,20 @@ export class LaunchesPage extends Component {
     deleteItemsAction: () => {},
   };
 
+  componentDidUpdate() {
+    clearInterval(this.intervalId);
+
+    this.arrIdLaunchInProgress = this.props.launches
+      .filter((item) => item.status === 'IN_PROGRESS')
+      .map((item) => item.id);
+
+    if (this.arrIdLaunchInProgress.length) {
```

Comment:
I think you should also check if launches list has changed. Otherwise, timeout will be reset even on launch selection.

```
if (/* check if we still have launches in progress and array has been changed and there's no active timeout (this.timeout === null) */) {
  // ...
}
```